### DISCUSSION TITLE
How to reduce unused Javascript from a Google Tag Script?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#66079] [Help] [answered]
### Summary

This is the error:
![image](https://github.com/vercel/next.js/assets/93617433/1f7b47f4-f30e-4f16-9c9a-2aad1a3973fe)

^^  This is one Google tag that I had inside `_document`  with the strategy of `beforeInteractive`, then I change the strategy to `worker`, but with the worker strategy it wouldn't work to do any tracking.

Right now the tag is in the `app.js ` with the strategy of `afterInteractive` and still giving me the error.

I am running the lighthouse test in incognito browser.

What could I do to fix this error?

**Comment by icyJoseph** [comment, marked as answer]
You've reached the end of the road. https://medium.com/@damupi/why-we-will-never-reduce-the-unused-javascript-of-gtm-2f2afc2520d6

I mean it is possible to do some counter measures, and even try to use web workers, but the sad reality is that, the problem is the tag itself

search a bit online, you'll find tons of people frustrated by this

**Reply by Elindo586** [reply]
Thanks... at least that tag is the only reason why I am seeing that error.

**Comment by AurelianSpodarec** [comment]
So no workarounds? Guess we'll never recieve 100/100 on performance ;(

**Comment by ullaskunder3** [comment]
really damm
I have 99% just becasue of this gtag

**Comment by ullaskunder3** [comment]
wait this seems working:

```
"use client";
import { useEffect, useState } from "react";
import Script from "next/script";

export default function GoogleAnalyticsDeferred() {
  const [loadGTM, setLoadGTM] = useState(false);

  useEffect(() => {
    let fired = false;
    const fire = () => {
      if (!fired) {
        fired = true;
        setLoadGTM(true);
        remove();
      }
    };
    // idle fallback
    const timer = setTimeout(fire, 5000);
    const remove = () => {
      window.removeEventListener("scroll", fire);
      window.removeEventListener("pointerdown", fire);
      window.removeEventListener("keydown", fire);
      clearTimeout(timer);
    };
    window.addEventListener("scroll", fire, { once: true, passive: true });
    window.addEventListener("pointerdown", fire, { once: true });
    window.addEventListener("keydown", fire, { once: true });
    return remove;
  }, []);

  if (!loadGTM) return null;

  return (
    <>
      <Script
        async
        id="gtag-js"
        src={`https://www.googletagmanager.com/gtag/js?id=${process.env
          .NEXT_PUBLIC_MEASUREMENT_ID!}`}
      />
      <Script id="gtag-init">{`
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', '${process.env.NEXT_PUBLIC_MEASUREMENT_ID!}');
      `}</Script>
    </>
  );
}

```

```tsx
import GoogleAnalyticsDeferred from "../components/script/GoogleAnalytics";

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang={"en"}>
      <GoogleAnalyticsDeferred />
```

**Reply by ullaskunder3** [reply]
I know not a good way to test but satisfying to want that 100% and confetti

<img width="844" height="666" alt="image" src="https://github.com/user-attachments/assets/2fd7c150-5a1f-4bc8-bbff-d0948aa6cafa" />
